Makefile

```diff
@@ -51,16 +51,29 @@ DSC_FILE_URL = "https://sonicstorage.blob.core.windows.net/packages/kernel-sourc
 DEBIAN_FILE_URL = "https://sonicstorage.blob.core.windows.net/packages/kernel-source/linux_4.9.168-1+deb9u5.debian.tar.xz?sv=2015-04-05&sr=b&sig=koCXHDvmY39smVGcI3cPJrBDZMmdpKiLkyJPfMdNPwU%3D&se=2033-04-28T06%3A14%3A51Z&sp=r"
 ORIG_FILE_URL = "https://sonicstorage.blob.core.windows.net/packages/kernel-source/linux_4.9.168.orig.tar.xz?sv=2015-04-05&sr=b&sig=ArvSGD3N46WGh%2BTYF8J1JgdT9x0BrFu4JhSuyyr3nNw%3D&se=2033-04-09T01%3A00%3A47Z&sp=r"
 
+ifneq ($(wildcard $(BUILD_DIR)),)
+ifeq ($(kernel_procure_method), incremental)
+INCR_BUILD = YES
+endif
+endif
+
 $(addprefix $(DEST)/, $(MAIN_TARGET)): $(DEST)/% :
 	# Obtaining the Debian kernel source
+ifeq ($(INCR_BUILD),)
 	rm -rf $(BUILD_DIR)
 	wget -O $(DSC_FILE) $(DSC_FILE_URL)
 	wget -O $(ORIG_FILE) $(ORIG_FILE_URL)
 	wget -O $(DEBIAN_FILE) $(DEBIAN_FILE_URL)
 
 	dpkg-source -x $(DSC_FILE)
+endif
 
 	pushd $(BUILD_DIR)
+
+ifneq ($(INCR_BUILD),)
+	# Force kernel build
+	rm -f debian/stamps/build_amd64_none_amd64
```

Comment:
in this incremetal build, it looks like we are not re-applying the patch series. what if the developer add a new kernel patch? Can you do stg pop and then re-apply all the patches?

Reply:
The theory is that, the user will first try the patch by directly applying the changes to the staged repo "sonic-linux-kernel/linux-4.9.189". Re-build, test the image.  In fact, to create the patch itself, the staged repo "sonic-linux-kernel/linux-4.9.189" shall be used and when all change set is ready the patches are moved to the patch directory and series file updated. The staged repo is transient in nature till the point patches are tried manually, tested and finalized.